app/src/main/java/de/westnordost/streetcomplete/settings/OAuthFragment.kt

```diff
@@ -164,7 +164,7 @@ class OAuthFragment : Fragment(R.layout.fragment_oauth),
 
         override fun shouldOverrideUrlLoading(view: WebView?, url: String?): Boolean {
             if (url == null) return false
-            val uri = Uri.parse(url) ?: return false
+            val uri = url.toUri()
```

Comment:
Hm right, this probably never worked. My intention here was that if the string is not a valid uri, it would just return false. Is there `String.toUriOrNull` or something?

Reply:
Nope.

Reply:
You should be able to replace 166 and 167 with
```kotlin
val uri = url?.takeIf(::isValidUrl)?.toUri() ?: return false
```

https://developer.android.com/reference/android/webkit/URLUtil#isValidUrl(java.lang.String)

Reply:
(Of course, then you need to import `android.webkit.URLUtil`)

Reply:
Thanks for the suggestion. I'll update the code.

Reply:
Actually, you can do the same with the lines after, which is another good readability increase, I think
```kotlin
val uri = url.takeIf(::isValidUrl)?.toUri().takeIf {
    it.isHierarchical && it.scheme == callbackScheme && it.host == callbackHost
} ?: return false
```
Although, personally control flow statements without brackets and newlines\* make me nervous, so I'd spend the vertical space for
```kotlin
if (uri == null) {
    return false
}
```
\*exception: one-liners like `return if (a) b else c`

Reply:
I suppose you could do
```kotlin
if (uri != null) {
    // verifier stuff
}
return uri == null
```
or if you don't give a damn about readability and only care about vertical space :rofl: 
```kotlin
return url.takeIf(::isValidUrl)?.toUri().takeIf {
    it.isHierarchical && it.scheme == callbackScheme && it.host == callbackHost
}?.also { uri -> continuation?.let { c ->
    uri.getQueryParameter(OAUTH_VERIFIER)?.let { verifier ->
        c.resume(verifier)
    } ?: c.resumeWithException(
        OAuthExpectationFailedException("$OAUTH_VERIFIER parameter not set by provider")
    )
}} != null
```